src/com/amplitude/api/AmplitudeClient.java

```diff
@@ -898,14 +891,21 @@ public void run() {
         long maxIdentifyId = -1;
 
         while (merged.length() < numEvents) {
+            boolean noEvents = events.isEmpty();
+            boolean noIdentifys = identifys.isEmpty();
+
+            // case 0: no events or identifys, nothing to grab
+            if (noEvents && noIdentifys) {
+                break;
```

Comment:
note that this case should never occur. if it does then it is an error, and should log a warning